simtools/layout/layout_array_builder.py

```diff
@@ -0,0 +1,244 @@
+        )
+
+        return telescopes
+
+    @u.quantity_input(x=u.m, y=u.m, radius=u.m)
+    def _get_telescope_patch(self, name, x, y, radius):
+        """
+        Collect the patch of one telescope to be plotted by self.plot_array.
+
+        Parameters
+        ----------
+        name: str
+            Name of the telescope (type).
+        x: astropy.units.m
+            X position of the telescope in meters.
+        y: astropy.units.m
+            Y position of the telescope in meters.
+        radius: astropy.units.m
+            Radius of the telescope sphere in meters.
+
+        Returns
+        -------
+        patch
+            Instance of mpatches.Circle.
+        """
+
+        valid_name = names.get_telescope_type(name)
+        fill_flag = False
+        if valid_name == "MST":
+            fill_flag = True
+        if valid_name == "SCT":
+            patch = mpatches.Rectangle(
+                ((x - radius / 2).value, (y - radius / 2).value),
+                width=radius.value,
+                height=radius.value,
+                fill=False,
+                color=self.colors_dict["SCT"],
+            )
+        else:
+            patch = mpatches.Circle(
+                (x.value, y.value),
+                radius=radius.value,
+                fill=fill_flag,
+                color=self.colors_dict[valid_name],
+            )
+        return patch
+
+    @staticmethod
+    @u.quantity_input(rotatio_angle=u.deg)
+    def _rotate(rotation_angle, x, y):
+        """
+        Rotate x and y by the rotation angle given in rotation_angle.
+        The function returns the rotated x and y values.
+        Notice that rotation_angle must be given in degrees!
+
+        Parameters
+        ----------
+        rotation_angle: astropy.units.deg
+            Angle to rotate the array in degrees.
+        x: numpy.array or list
+            X positions of the telescopes, usually in meters.
+        y: numpy.array or list
+            Y positions of the telescopes, usually in meters.
+
+        Returns
+        -------
+        2-tuple of list
+            X and Y positions of the rotated telescopes positions.
+
+        """
+        if not isinstance(x, type(y)):
+            raise RuntimeError("x and y are not the same type! Cannot perform transformation.")
+        if not isinstance(x, (list, np.ndarray)):
+            x = [x]
+            y = [y]
+
+        if len(x) != len(y):
+            raise RuntimeError(
+                "Cannot perform coordinate transformation when x and y have different lengths."
+            )
+
+        try:
+            x = x.to(u.m)
+            y = y.to(u.m)
+            if isinstance(x[0].unit, type(u.m)) and isinstance(y[0].unit, type(u.m)):
+                x = x.value
+                y = y.value
+        except AttributeError:
+            pass
+
+        x_trans, y_trans = [], []
+        rotate_angle = np.deg2rad(rotation_angle.value)
```

Comment:
I recommend using `to('rad')` instead. You don't have to enforce degrees as well in this case.